src/DynamoPackages/PackageManagerExtension.cs

```diff
@@ -120,7 +123,33 @@ public void Startup(StartupParams startupParams)
             LoadPackages(startupParams.Preferences, startupParams.PathManager);
         }
 
-        public void Ready(ReadyParams sp) { }
+        public void Ready(ReadyParams sp)
+        {
+            sp.CurrentWorkspaceChanged += OnCurrentWorkspaceChanged;
```

Comment:
these events and the ones below should be unsubscribed when those workspaces are no longer the current workspace or the workspaces will not be able to be garbage collected -- not sure about there logic there in all cases, ie what about custom node workspaces?

We should also implement cleanup in the shutdown or dispose method of this Extension, but be careful as there is an inconsistency between shutdown and dispose between viewExtension and Extensions - I can't remember which is really called on extensions 😢 - please check.

Reply:
I believe this is addressed now